document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/vistors/PostgresAggregationFilterTypeExpressionVisitor.java

```diff
@@ -22,27 +29,22 @@ public String visit(final RelationalExpression expression) {
     SelectTypeExpression rhs = expression.getRhs();
 
     // Only an identifier LHS and a constant RHS is supported as of now.
-    PostgresSelectTypeExpressionVisitor lhsVisitor = new PostgresIdentifierExpressionVisitor();
-    PostgresSelectTypeExpressionVisitor rhsVisitor = new PostgresConstantExpressionVisitor();
-
     String key = lhs.accept(lhsVisitor);
     Object value = rhs.accept(rhsVisitor);
 
-    // In SQL, the alias is not supported in the Filter and Having clause.
-    // As of now, where is clause is first parsed and it is not using any alias expression.
-    // However, having clause is parsed after group by, and currently, it is only
-    // supported with alias expression.
-
-    if (!postgresQueryParser.getPgSelections().containsKey(key)) {
-      throw new UnsupportedOperationException(
-          "Having clause is only supported with alias expression"
-              + "for aggregation and functional expression");
+    if (postgresQueryParser.getPgSelections().containsKey(key)) {
```

Comment:
Removed the restriction added to move on to allow only functional expressions as part of having clause.